src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -1448,6 +1448,28 @@ class LedgerMasterImp
 
     Ledger::pointer getLedgerBySeq (std::uint32_t index)
     {
+        if (index <= mValidLedgerSeq)
+        {
+            // Always prefer a validated ledger
+            auto valid = mValidLedger.get ();
```

Comment:
Since you only use valid in this scope, consider:

`if (auto valid = = mValidLedger.get ())`